### PR TITLE
Revise BatchNormInference Reference Implementation

### PR DESCRIPTION
**Pull request by ggalieroc**
BatchNormInference reference implementation refactored.

Details:
- Operation class review
- Refactored type_prop tests to cover validation checks for v0 and v5 of operation
- Refactored backend tests. Removed instances for f64 precision and epsilon attribute value zero
- Removed legacy code for training and back propagation
- Modified signature of ref impl to align with v5
- Added serialization single layer tests
- Added visitor attribute tests

Ticket: 37423

### CODE REVIEW

File: ngraph/test/visitors/op/batch_norm.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    auto gamma = make_shared<op::Parameter>(et, ch_shape);
+    auto beta = make_shared<op::Parameter>(et, ch_shape);
+    auto mean = make_shared<op::Parameter>(et, ch_shape);
+    auto var = make_shared<op::Parameter>(et, ch_shape);
+    auto batch_norm = make_shared<TypeParam>(data_batch, gamma, beta, mean, var, epsilon);
+
+    NodeBuilder builder(batch_norm);
+    auto g_batch_norm = as_type_ptr<TypeParam>(builder.create());
+    EXPECT_EQ(g_batch_norm->get_eps_value(), batch_norm->get_eps_value());
+}
```
Comment (pelszkow) [inline ngraph/test/visitors/op/batch_norm.cpp:46]: IMHO, we should add similar to [this](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/visitors/op/squeeze.cpp#L25-L27) check for each visitor test.
